weave

```diff
@@ -741,8 +752,13 @@ setup_cni() {
     if install_cni_plugin $CNI_PLUGIN_DIR ; then
         upgrade_cni_plugin $CNI_PLUGIN_DIR
     fi
-    if [ -d $HOST_ROOT/etc/cni/net.d -a ! -f $HOST_ROOT/etc/cni/net.d/10-weave.conf ] ; then
-        create_cni_config $HOST_ROOT/etc/cni/net.d/10-weave.conf $HAIRPIN_MODE
+
+    # Weave will use .conflist for versions > 2.4.0. So in case of upgrade of old
+    # clusters with Weave version <= 2.4.0 remove existing 10-weave.conf
+    rm -f $HOST_ROOT/etc/cni/net.d/10-weave.conf
```

Comment:
I really don't like an `rm -f` on $HOST_ROOT in a script running in a container. I want the feature to be implemented so that we can stop putting a `00-weave.conflist` on the hosts via SaltStack but there's got to be an opt-in way for old clusters. Leave it alone and add a `09-weave.conflist` so as to not break other people's overrides?

Reply:
It would be ideal to avoid supporting both `.conf` and `.conflist` so that both new and existing deployments works in consistent manner. Also note that other CNI's (calico, flannel for e.g.) already supports `.conflist` only. Since the change explicitly deleting a single file should that be really of that concern? What could be possible issue doing so?